conjure-codegen/src/context.rs

```diff
@@ -1002,6 +1027,84 @@ impl Context {
         }
     }
 
+    fn type_log_safety(&self, ty: &Type) -> Option<LogSafety> {
```

Comment:
This logic should mirror what's defined in https://github.com/palantir/conjure-java/blob/develop/conjure-java-core/src/main/java/com/palantir/conjure/java/types/SafetyEvaluator.java.